src/components/select-field/SelectField.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+
+import * as React from 'react';
+import getProp from 'lodash/get';
+import type { FieldProps } from 'formik';
+
+import SingleSelectPrimitive from './SingleSelectField';
+import MultiSelectPrimitive from './MultiSelectField';
+
+import type { SelectOptionValueProp, SelectOptionProp } from './props';
+
+type Props = {
+    multiple?: boolean,
+} & FieldProps;
+
+function createFakeSyntheticEvent(name: string, value: SelectOptionValueProp | Array<SelectOptionValueProp>) {
+    return {
+        currentTarget: { name, value },
+        target: { name, value },
+    };
+}
+
+function onSelect(name: string, onChange: Function, options: SelectOptionProp | Array<SelectOptionProp>) {
+    if (Array.isArray(options)) {
```

Comment:
Minor, but it seems like this could live as a ternary to avoid duplicating the `onChange` calls.